### PR TITLE
fix: useSearch change to handle modal search gf-455

### PR DESCRIPTION
**Pull request by Fjortis**
This change is implemented to avoid unintended double retendering when using the useSearch where search queries aren't relevant. 
Improved flexibility: Allows the search functionality to be tailored for various contexts (e.g., modal or page search) without hardcoding specific behaviors.

### CODE REVIEW

File: apps/frontend/src/libs/hooks/use-search/use-search.hook.ts [hunk 1]
```diff
@@ -1,20 +1,26 @@
 import { QueryParameterName } from "~/libs/enums/enums.js";
 import { useEffect, useSearchParams, useState } from "~/libs/hooks/hooks.js";
 
-const useSearch = (): {
+const useSearch = (
+	useQueryParameters: boolean = true,
+): {
```
Comment (perekotypole) [inline apps/frontend/src/libs/hooks/use-search/use-search.hook.ts:6]: I think it's better to pass an argument as an object, useSeatch(false) doesn't look very clear. 
And don't forget about B19 (now naming is the same as hooks)

**Comment by Fjortis** [comment]
I am unsure about this change, perhaps it is better to make use of useState and useAppForm for modals instead of using useSearch. Do you think this is a better way of doing it?

File: apps/frontend/src/pages/access-management/libs/components/group-users-table/group-users-table.tsx [hunk 2]
```diff
@@ -43,7 +43,7 @@ const GroupUsersTable = ({
 	setValue,
 }: Properties): JSX.Element => {
 	const dispatch = useAppDispatch();
-	const { onSearch, search } = useSearch();
+	const { onSearch, search } = useSearch({ shouldUseQuery: false });
```
Comment (liza-veis) [inline apps/frontend/src/pages/access-management/libs/components/group-users-table/group-users-table.tsx:46]: let's rename it to isSavedToUrl
```suggestion
	const { onSearch, search } = useSearch({ isSavedToUrl: false });
```

File: apps/frontend/src/libs/hooks/use-search/use-search.hook.ts [hunk 3]
```diff
@@ -1,20 +1,35 @@
 import { QueryParameterName } from "~/libs/enums/enums.js";
 import { useEffect, useSearchParams, useState } from "~/libs/hooks/hooks.js";
 
-const useSearch = (): {
+type SearchOptions = {
```
Comment (liza-veis) [inline apps/frontend/src/libs/hooks/use-search/use-search.hook.ts:4]:
```suggestion
type Parameters = {
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]
lgtm after Lisa's changes

**Review by liza-veis — APPROVED** [review]

**Pull request merged**